### PR TITLE
fix(ingestion): powerbi # continue ingestion if m-query parsing fail

### PR DESCRIPTION
**Pull request by mohdsiddique**

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py [hunk 1]
```diff
@@ -120,11 +123,11 @@ def _get_entity_users(
                 entity=entity_name,
                 entity_id=entity_id,
             )
-        except Exception as e:
-            self.log_http_error(
-                message=f"Unable to fetch users for {entity_name}({entity_id}).", e=e
+        except:  # It will catch all type of exception
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py:126]: What was the problem with the `Exception as e` way of catching exception handling?

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/parser.py [hunk 2]
```diff
@@ -53,22 +54,30 @@ def get_upstream_tables(
...
+        return resolver.MQueryResolver(
+            table=table,
+            parse_tree=parse_tree,
+            reporter=reporter,
+        ).resolve_to_data_platform_table_list()  # type: ignore
+
+    except:  # noqa: E722
+        # It will catch all type of exceptions, so that ingestion can continue without lineage information
+        _, e, _ = sys.exc_info()
+        logger.warning(str(e))
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/parser.py:75]: Thank you- we need MORE of this.

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py [hunk 3]
```diff
@@ -171,9 +171,12 @@ def _process_invoke_expression(
...
 
-        first_argument: Tree = tree_function.flat_argument_list(first_arg_tree)[
-            0
-        ]  # take first argument only
+        flat_arg_list: List[Tree] = tree_function.flat_argument_list(first_arg_tree)
+        if len(flat_arg_list) == 0:
+            logger.debug("flat_arg_list is zero")
+            return None
+
+        first_argument: Tree = flat_arg_list[0]  # take first argument only
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py:179]: Thank you for adding this check. We need MORE of this!

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py (hunk 1 above)
Comment (jjoyce0510) [reply metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py:126]: +1 - I'm somewhat confused

Comment (mohdsiddique) [reply metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py:126]: It is not catching ConnectionError, so Exception is not good if you want to catch all types of exceptions.

Comment (jjoyce0510) [reply metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py:126]: Ah - thanks for clarification. This is useful to know!

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py [hunk 4]
```diff
@@ -207,12 +210,11 @@ def _get_scan_result(self, workspace: Workspace) -> Any:
...
-        except Exception as e:
-            self.log_http_error(
-                message=f"Unable to fetch dataset lineage for {workspace.name}({workspace.id}).",
-                e=e,
+        except:
+            e = self.log_http_error(
+                message=f"Unable to fetch dataset lineage for {workspace.name}({workspace.id})."
             )
-            if data_resolver.is_permission_error(e):
+            if data_resolver.is_permission_error(cast(Exception, e)):
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py:217]: Is this a safe cast if e is not actually an exception?

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py [hunk 5]
```diff
@@ -183,8 +186,8 @@ def get_workspaces(self) -> List[Workspace]:
         groups: List[dict] = []
         try:
             groups = self._get_resolver().get_groups()
-        except Exception as e:
-            self.log_http_error(message="Unable to fetch list of workspaces", e=e)
+        except:
+            self.log_http_error(message="Unable to fetch list of workspaces")
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py:190]: Won't we be losing this exception?

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py [hunk 6]
```diff
@@ -147,9 +150,9 @@ def get_reports(self, workspace: Workspace) -> List[Report]:
         reports: List[Report] = []
         try:
             reports = self._get_resolver().get_reports(workspace)
-        except Exception as e:
+        except:
             self.log_http_error(
-                message=f"Unable to fetch reports for workspace {workspace.name}", e=e
+                message=f"Unable to fetch reports for workspace {workspace.name}"
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py:155]: Won't we be losing this exception?

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py [hunk 7]
```diff
@@ -120,11 +123,11 @@ def _get_entity_users(
...
             )
-        except Exception as e:
-            self.log_http_error(
-                message=f"Unable to fetch users for {entity_name}({entity_id}).", e=e
+        except:  # It will catch all type of exception
+            e = self.log_http_error(
+                message=f"Unable to fetch users for {entity_name}({entity_id})."
             )
-            if data_resolver.is_permission_error(e):
+            if data_resolver.is_permission_error(cast(Exception, e)):
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py:130]: Is this cast safe? Given that e may not be an exception

**Review by jjoyce0510 — APPROVED** [review]
Want to get these changes for error handling out. Let's follow up on the dropping exceptions piece offline!

Cheers
John

**Pull request merged**